### PR TITLE
[ Release-1.21 ] - remove socat from dockerfile (#2210)

### PR DESCRIPTION
**Pull request by briandowns**
Backport of issue #2208

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -113,16 +113,12 @@ RUN rm -vf /charts/*.sh /charts/*.md
 # This image includes any host level programs that we might need. All binaries
 # must be placed in bin/ of the file image and subdirectories of bin/ will be flattened during installation.
 # This means bin/foo/bar will become bin/bar when rke2 installs this to the host
-FROM rancher/k3s:v1.21.7-k3s1 AS k3s
-FROM rancher/hardened-kubernetes:v1.21.7-rke2r1-build20211119 AS kubernetes
-FROM rancher/hardened-containerd:v1.4.12-k3s1-build20211118 AS containerd
-FROM rancher/hardened-crictl:v1.21.0-build20211118 AS crictl
+FROM rancher/hardened-kubernetes:v1.22.4-rke2r1-build20211119 AS kubernetes
```
Comment (galal-hussein) [inline Dockerfile:116]: why are we upgrading kubernetes to 1.22?

File: Dockerfile [hunk 2]
```diff
@@ -113,16 +113,12 @@ RUN rm -vf /charts/*.sh /charts/*.md
...
+FROM rancher/hardened-containerd:v1.5.8-k3s1-build20211119 AS containerd
+FROM rancher/hardened-crictl:v1.22.0-build20211118 AS crictl
```
Comment (dweomer) [inline Dockerfile:118]: these need to be adjust for 1.21.x, amirite?
```
FROM rancher/hardened-kubernetes:v1.22.4-rke2r1-build20211119 AS kubernetes
FROM rancher/hardened-containerd:v1.5.8-k3s1-build20211119 AS containerd
FROM rancher/hardened-crictl:v1.22.0-build20211118 AS crictl
```

**Review by dweomer — CHANGES_REQUESTED** [review]
pulled in wrong versions with the cherry-pick it would seem

File: Dockerfile [hunk 3]
```diff
@@ -113,16 +113,11 @@ RUN rm -vf /charts/*.sh /charts/*.md
 # This image includes any host level programs that we might need. All binaries
 # must be placed in bin/ of the file image and subdirectories of bin/ will be flattened during installation.
 # This means bin/foo/bar will become bin/bar when rke2 installs this to the host
-FROM rancher/k3s:v1.21.7-k3s1 AS k3s
 FROM rancher/hardened-kubernetes:v1.21.7-rke2r1-build20211119 AS kubernetes
 FROM rancher/hardened-containerd:v1.4.12-k3s1-build20211118 AS containerd
 FROM rancher/hardened-crictl:v1.21.0-build20211118 AS crictl
-FROM rancher/hardened-runc:v1.0.2-build20211118 AS runc
```
Comment (brandond) [inline Dockerfile:120]: why is runc going away? Doesn't this break the dockerfile, as we still `COPY --from=runc`?

**Review by brandond — APPROVED** [review]

**Review by dweomer — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]

**Pull request merged**